### PR TITLE
Added `AccountClaim` message

### PR DESCRIPTION
**Pull request by lucasvo**
It is not always possible to verify & link account ownership across chains. This PR proposes a new message type that allows a chain to send a message to another chain indicating that the owner of an account successfully authenticated itself on the chain and intended to make such a claim on the target chain. This can be used to address an issue with the current crowdloan pallet discussed in [polkadot/#2311](https://github.com/paritytech/polkadot/issues/2311)

When thinking about this problem a bit, I arrived at the conclusion that this is essentially a "claim" an identity or account can make. At the very minimum I think such a claim must have some sort of "scope" and an expiration date allowing the claims to be protected against typical issues such as replay protection and front-running.

I think there is a case to be made to support both vertical message passing as well as horizontal message passing. We should be able to use the same format for both, I assume right?

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -222,7 +223,19 @@ Remove the asset(s) (`assets`) from holding and send a `WithdrawAsset` XCM messa
...
 
+<<<<<<< HEAD
 Errors:
+=======
+### `AccountClaim`
+Sends a claim verified to be from the `source` to the `destination` on the parachain. It has an `expiration` timestamp and a `scope` used to define the scope under which such a `claim` is made. A chain should only send such a claim if it is verified to have been intended to be sent by the `source`. It facilitates authentication across parachains.
+
+- `source: MultiLocation`
+- `destination: MultiLocation`
+- `timestamp: Compact<u48>`
```
Comment (xlc) [inline README.md:234]: How is timestamp going to be used? Upon receive the message, the dest chain should just transfer the assets into the destination account immediately.

Comment (lucasvo) [reply README.md]: This is not used for assets but merely indicates the validity of the claim. In the crowdloan example it would you would first sign an extrinsic on the relay chain instructing it to send a claim to the parachain with such a claim and an expiration date. Until such a timestamp has passed, the parachain would allow the user to then submit an extrinsic to withdraw their reward from the `destination` account.

File: README.md [hunk 2]
```diff
@@ -173,6 +174,16 @@ Kind: *Trusted Indication*.
 
 Errors:
 
+### `AccountClaim`
+Sends a claim verified to be from the `source` to the `destination` on the parachain. It has an `expiration` timestamp and a `scope` used to define the scope under which such a `claim` is made. A chain should only send such a claim if it is verified to have been intended to be sent by the `source`. It facilitates authentication across parachains.
+
+- `source: MultiLocation`
+- `destination: MultiLocation`
+- `expiration: Compact<u48>`: Timestamp of the claim expiration
```
Comment (gavofyork) [inline README.md:182]: when exactly would this be used?

File: README.md [hunk 3]
```diff
@@ -173,6 +174,16 @@ Kind: *Trusted Indication*.
 
 Errors:
 
+### `AccountClaim`
+Sends a claim verified to be from the `source` to the `destination` on the parachain. It has an `expiration` timestamp and a `scope` used to define the scope under which such a `claim` is made. A chain should only send such a claim if it is verified to have been intended to be sent by the `source`. It facilitates authentication across parachains.
+
+- `source: MultiLocation`
```
Comment (gavofyork) [inline README.md:180]: how does this differ from `origin`?

File: README.md [hunk 4]
```diff
@@ -173,6 +174,16 @@ Kind: *Trusted Indication*.
...
+- `destination: MultiLocation`
+- `expiration: Compact<u48>`: Timestamp of the claim expiration
+- `scope: Vec<u8>`
```
Comment (gavofyork) [inline README.md:183]: how is this defined?

**Comment by h4x3rotab** [comment]
Looks like a `Transact` is sufficient to send a `Call` to the targeted account. IMHO "AccountClaim" is too specific use case to fit into XCM itself. I'd love to keep XCM more generic.

**Comment by lucasvo** [comment]
A transact would work indeed @h4x3rotab, if the plan is for the `origin_type` to be always the origin account on the chain sending the message. The spec is a bit hard to follow here.

File: README.md (hunk 2 above)
Comment (lucasvo) [reply README.md:182]: Any certificate or claim should have an expiration IMO. Just looking at how signed statements would be used when working in different contexts. For aanything from SSL to meta-transactions in crypto, most often you do need some sort of expiry or nonce but nonces wouldn't make sense in this instance.

File: README.md (hunk 3 above)
Comment (lucasvo) [reply README.md:180]: Various, let's assume you use a claim to authorize an account to execute trades on behalf of it one a target chain, you could use the scope to limit it to a certain currency.

File: README.md (hunk 4 above)
Comment (lucasvo) [reply README.md:183]: For example if a claim was used to authorize another account to trade you would create a claim to the dex as destination and set the scope to the assets you would authorize a transfer for.

**Comment by xlc** [comment]
I will prefer specific claim XCM over generic Transact. 

With an universally implemented XCM, we can offer an unified flow to handle claims. It should work on all chains and sender does not need to worry about dest chain extrinsic format, which can change. 

Especially for ledger or other hardware devices, it will be impossible for the device to correctly parse & show the payload of a Transact message.

**Comment by gavofyork** [comment]
@lucasvo / @xlc is this still needed? If so it will need to be resolved.

**Comment by xlc** [comment]
I still think this is a useful feature to have. It could also be used for example connecting identities across multiple parachains.

File: README.md (hunk 3 above)
Comment (lucasvo) [reply README.md:180]: I'll remove source as it is indeed the same as `origin`

File: README.md (hunk 2 above)
Comment (lucasvo) [reply README.md:181]: @gavofyork are you ok with leaving this in?

**Comment by lucasvo** [comment]
@xlc perhaps it would make sense to ask the Kilt team on this as well.
